cmake/sca/codechecker/sca.cmake

```diff
@@ -8,9 +8,14 @@ message(STATUS "Found CodeChecker: ${CODECHECKER_EXE}")
 # CodeChecker uses the compile_commands.json as input
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-# Create an output directory for our tool
-set(output_dir ${CMAKE_BINARY_DIR}/sca/codechecker)
-file(MAKE_DIRECTORY ${output_dir})
+if (DEFINED CODECHECKER_OUTPUT_DIR)
+  # Create an output directory for our tool
+  set(output_dir ${CODECHECKER_OUTPUT_DIR})
```

Comment:
build system should not start to allow users to locate files create during build outside the build folder.

Reply:
The whole point of this change was to allow our CI system to put the codechecker results where it's convenient. In that case I'll close the PR.